ext_mongodb.php

```diff
@@ -715,17 +720,43 @@ function __debugInfo() : array;
 {
 	use DenySerialization;
 
-	public function __construct(private string $code, private ?mixed $scope = NULL)
+	private $code;
+
+	public function __construct(string $code, private ?mixed $scope = NULL)
 	{
+		if ( strstr( $code, "\0" ) !== false )
+		{
+			throw new \MongoDB\Driver\Exception\InvalidArgumentException( "Code cannot contain null bytes" );
+		}
+		$this->code = $code;
 	}
 
 	public function __debugInfo() : array
 	{
 		return [
-			'javascript' => $this->code,
+			'code' => $this->code,
 			'scope' => (object) $this->scope
 		];
 	}
+
+	public function getCode() : string
+	{
+		return $this->code;
+	}
+
+	public function getScope() : mixed
+	{
+		if ( $this->scope !== NULL )
+		{
+			return (object) $this->scope;
```

Comment:
I gather this is probably done because PHPC is round-tripping `$scope` through BSON, and a top-level array would be changed to an object (see: [this test](https://github.com/mongodb/mongo-php-driver/blob/master/tests/bson/bson-javascript-getScope-001.phpt)). Is it really worth the casting, or would you rather skip or alter the test?

Note that the drivers might still yield different results if a variable within the scope was itself a document. PHPC would return it as an object but you'd still return an array.

Sub-note: I realize that scope variables are supposed to be scalars, but MongoDB still supports complex values and users do weird things.